src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -249,11 +250,9 @@ public boolean perform(AbstractBuild<?, ?> build,
                         //git.push(null);
                     }
                 } catch (FormException e) {
-                    e.printStackTrace(listener.error("Failed to push merge to origin repository"));
-                    return false;
+                    throw new AbortException("Failed to push merge to origin repository.\n" + e.getMessage());
                 } catch (GitException e) {
-                    e.printStackTrace(listener.error("Failed to push merge to origin repository"));
-                    return false;
+                    throw new AbortException("Failed to push merge to origin repository\n" + e.getMessage());
```

Comment:
This (and some other similar cases) seems like a regression in diagnosability. I do not know exactly what the stack trace in the `GitException` might be, but it might be meaningful. If it is guaranteed to only arise from a bounded set of user errors, where the `.message` explains everything, then fine. But what if the `GitException.cause` is some `SocketException` wrapped by JGit, etc.?